crates/transpiler/src/passes/vf2/vf2_layout.rs

```diff
@@ -347,10 +347,14 @@ fn map_free_qubits(
             .get(&[PhysicalQubit::new(*qubit_b), PhysicalQubit::new(*qubit_b)])
             .unwrap_or(&0.);
         // Reverse comparison so lower error rates are at the end of the vec.
-        score_b.partial_cmp(&score_a).unwrap()
+        match score_b.partial_cmp(&score_a).unwrap() {
+            Ordering::Equal => qubit_b.cmp(qubit_a),
+            Ordering::Less => Ordering::Less,
+            Ordering::Greater => Ordering::Greater,
+        }
```

Comment:
No need to change what's written, but if you like: `Ordering` has a helper method for tuple-ish comparisons, which is `Ordering::then` (or `then_with`), so
```rust
score_b.partial_cmp(&score_a).expect("no scores are nan").then(qubit_b.cmp(qubit_a))
```
is possible too.